# Diary

* 2018/2/1
    * 第七章範例練習完了,可是7-24下面的範例測試不會寫|||
    * 今天又吃了三媽,然後秀秀又帶了起司蛋糕回來,覺得幸福,雖然我好像又要變胖了 囧
    * 最近有點混啊,明天先看完第八章再來思考實作題怎麼寫吧

* 2018/2/2
    * 廖紫嵐當天來回台中,最後被他家吳先生載回家了,然後姊姊也回家了
    * 量體重真的是胖了......
    * 看了第八章,有點不懂....例外處理的架構什麼的,突然講得有點複雜

* 2018/2/3
    * 晚餐吃了火鍋,覺得滿足!!!
    * 看了柯南劇場版異次元的狙擊手...劇情不錯,不過演到後來變成動作片了哈哈哈
    * 第八章例外處理姑且看完了,不過大概只有半懂的程度
    * Java中,所有錯誤都會被包裹為物件,為java.lang.Throwable的各種子類別實例
    * 可以嘗試(try)執行程式並捕捉(catch)代表錯誤的物件後作一些處理,也可以將物件拋出(throws)
    * Throwable定義了取得錯誤訊息.堆疊追蹤(Stack Trace)等方法,它有兩個子類別: java.lang.Error 與 java.lang.Exception
    * Error與其子類別實例代表嚴重系統錯誤,例如硬體層面錯誤或記憶體不足等問題,基本上Java應用程式本身是無力處理的
    * 程式設計本身的錯誤,建議使用Exception或其子類別實例來表現,所以通常稱錯誤處理為例外處理(Exception handling)
    * Exception或其子物件,但非屬於RuntimeException或其子物件,稱為受檢例外(Checked Exception)(受編譯器檢查),引發錯誤的時候,認為客戶端有能力處理錯誤
    * 屬於RuntimeException(Exception的子類別之一)衍生出來的類別實例,代表API設計者實作某方法時,某些條件成立時會引發錯誤,應在呼叫方法前做好檢查,以避免引發錯誤,之所以命名為執行時期例外,是因為編譯器不會強迫得處理,亦稱為非受檢例外(Unchecked Exception)
    * 若父類別例外物件在子類別例外物件前被捕捉,則catch子類別例外物件區塊將永遠不會被執行,但JDK7開始可以使用多重捕捉(multi-cath)語法,不過例外不得有繼承關係
    * 在catch區塊進行玩部分的錯誤處理後,可以使用throw(不是throws)將例外在拋出
    * 若想知道例外發生的根源,以及多重方法呼叫下例外的堆疊傳播,可以利用printStackTrace().getStackTrace()等方法
    * 無論try區塊中有無發生例外,若撰寫有finally區塊,則finally區塊一定會被執行
    * JDK7之後新增了嘗試關閉資源(try-with-resources)語法,想要嘗試自動關閉資源的的物件,是撰寫在try之後的括號中
    * 嘗試關閉資源語法可套用的物件,必需實作java.lang.AutoCloseable介面

* 2018/2/4
    * 晚餐吃了羊肉爐YA!
    * 終於把シュヴァリエ補完了...虐呀